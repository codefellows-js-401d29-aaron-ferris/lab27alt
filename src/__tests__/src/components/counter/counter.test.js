'use strict';
import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '../../../../components/counter/counter.js';

describe('<Counter />', () => {
  it('will run with stable DOM', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
describe('Down Event', () => {
  it('will change state.counter on a click', () => {
    let app = mount(<Counter />);
    let a = app.find('.down');
    let number = app.state('count');
    a.simulate('click');
    expect(app.state('count')).toEqual(number - 1);
  });

  it('will transfer state to the DOM', () => {
    let app = mount(<Counter />);
    let a = app.find('.down');
    let number = app.state('count');
    a.simulate('click');
    expect(app.find('span').text()).toContain(number - 1);
  });

});
describe('Up Event', () => {
  it('will change state.counter by +1', () => {
    let app = mount(<Counter />);
    let a = app.find('.up');
    let number = app.state('count');
    a.simulate('click');
    expect(app.state('count')).toEqual(number + 1);    
  });

  it('will transfer state to the DOM', () => {
    let app = mount(<Counter />);
    let a = app.find('.up');
    let number = app.state('count');
    a.simulate('click');
    expect(app.find('span').text()).toContain(number + 1);
  });

});



