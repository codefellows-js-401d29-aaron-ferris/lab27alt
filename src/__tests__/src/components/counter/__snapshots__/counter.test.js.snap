// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> will run with stable DOM 1`] = `
<section
  className="counter"
>
  <a
    className="down clicker"
    href="#"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="up clicker"
    href="#"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;

exports[`Down Event will run with stable DOM 1`] = `
<section
  className="counter"
>
  <a
    className="down clicker"
    href="#"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="up clicker"
    href="#"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;

exports[`Down Event will trigger with stable DOM 1`] = `
<section
  className="counter"
>
  <a
    className="down clicker"
    href="#"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="up clicker"
    href="#"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
